apps/admin_api/lib/admin_api/v1/controllers/configuration_controller.ex

```diff
@@ -37,13 +39,51 @@ defmodule AdminAPI.V1.ConfigurationController do
   def update(conn, attrs) do
     with {:ok, _} <- authorize(:update, conn.assigns),
          attrs <- put_originator(conn, attrs),
-         {:ok, settings} <- Config.update(attrs) do
+         {attrs, error} <- validate_master_account(attrs),
+         {:ok, settings} <- Config.update(attrs),
+         settings <- add_master_account_error(settings, error) do
       render(conn, :settings_with_errors, %{settings: settings})
     else
       {:error, code} -> handle_error(conn, code)
     end
   end
 
+  defp validate_master_account(%{"master_account" => master_account_id} = attrs) do
+    case Account.get(master_account_id) do
+      nil ->
+        {
+          Map.delete(attrs, "master_account"),
+          {
+            :master_account,
+            {:error,
+             %Changeset{
```

Comment:
```elixir
%StoredSetting{}
|> Changeset.change(%{value: master_account_id})
|> Changeset.apply_action(:update)
|> Changeset.add_error(:value, "must match an existing account", validation: :account_not_found)
```

... so we don't need to handle future Ecto's implementation changes.

Also updated `validation: master_account_id` to `validation: :account_not_found` so it aligns with other `add_error/4` usage (`validation: :validation_atom`).

Reply:
But using `validation: :account_not_found` returns 

```
 %{"code" => "client:invalid_parameter", "description" => "Invalid parameter provided. `value` must match an existing account.", "object" => "error", "messages" => %{"value" => ["account_not_found"]}}
```

While I wanted the `"value"` to have the actual value that was sent by the user 🤔

Reply:
The other validators already interpolated value into the error description string though: https://github.com/omisego/ewallet/blob/v1.2/apps/ewallet_config/lib/ewallet_config/setting_validator.ex